### PR TITLE
[WIP] Use JDK14 switch expressions - checkstyle needs to be updated

### PR DESCRIPTION
**Pull request by koppor**
I applied some IntelliJ suggestions - also includes using JDK14 switch expressions. This causes issues at checkstyle.

### CODE REVIEW

**Comment by koppor** [comment]
It was a "drive-by-shooting" while I was working on something else. Further, the wish was to keep pull requests small to easy reviewing (@stefan-kolb). Therefore, I spent effort to separate the easy-to-review things from another PR (not yet opened). This helps

- speed up reviewing
- speed up getting changes into `master`
- keep PRs small (if possible)

**Comment by koppor** [comment]
Blocked by https://github.com/checkstyle/checkstyle/issues/7992.

File: src/main/java/org/jabref/logic/layout/format/AuthorAndsReplacer.java [hunk 1]
```diff
@@ -21,24 +21,18 @@ public String format(String fieldText) {
...
-            s = authors[0] + " & " + authors[1];
-            break;
-        default:
-            int i;
-            int x = authors.length;
-            StringBuilder sb = new StringBuilder();
-
-            for (i = 0; i < (x - 2); i++) {
-                sb.append(authors[i]).append("; ");
+            case 1 -> s = authors[0]; // just no action
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/layout/format/AuthorAndsReplacer.java]: why not use the full power of the switch expressions and write `String s = switch (...)` ?

Comment (koppor) [reply src/main/java/org/jabref/logic/layout/format/AuthorAndsReplacer.java]: IntelliJ did not recommend it ^^. Just submit an update here :)

**Comment by koppor** [comment]
ETA for the fix [September 2020](https://github.com/checkstyle/checkstyle/issues/7992#issuecomment-605986115).

**Comment by koppor** [comment]
Closed to reduce the noise.

**Pull request closed without merging**